### PR TITLE
Fix part of #59: Adding Bazel build to CI checks [BLOCKED:  #1634]

### PR DESCRIPTION
**Pull request by miaboloix**
## Explanation
Fix part of #59: Adding Bazel build to CI checks [BLOCKED:  #1634]

This PR introduces a CI check that builds Oppia Android's apk with Bazel. 

Please note that this PR is replacing [this one](https://github.com/oppia/oppia-android/pull/1673).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -130,5 +130,44 @@ jobs:
...
+      matrix:
+        os: [ubuntu-18.04]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup bazel
+        uses: jwlawson/actions-setup-bazel@v1
+        with:
+          bazel-version: '3.4.1'
+      - name: Clone Oppia Bazel
+        run: git clone https://github.com/miaboloix/bazel.git $HOME/oppia-bazel
```
Comment (BenHenning) [inline .github/workflows/main.yml]:
```suggestion
        run: git clone https://github.com/oppia/bazel.git $HOME/oppia-bazel
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -130,5 +130,44 @@ jobs:
...
+
+  bazel_build_app:
+    name: Build Binary with Bazel
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup bazel
```
Comment (BenHenning) [inline .github/workflows/main.yml]:
```suggestion
      - name: Set up bazel
```

(set up is a verb vs noun setup)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -130,5 +130,44 @@ jobs:
...
+        run: git clone https://github.com/miaboloix/bazel.git $HOME/oppia-bazel
+      - name: Set up JDK 9
+        uses: actions/setup-java@v1
+        with:
+          java-version: 9
+      - name: Create tmp/android_tools
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/tmp/android_tools
+      - name: Extract Android Tools
+        run: |
```
Comment (BenHenning) [inline .github/workflows/main.yml]:
```suggestion
      - name: Extract Android Tools
        run: |
          mkdir -p $GITHUB_WORKSPACE/tmp/android_tools
```

Probably can collapse these two for simplicity.

**Review by BenHenning — APPROVED** [review]
Thanks @miaboloix. LGTM!

**Review by BenHenning — CHANGES_REQUESTED** [review]
Actually, I found one issue. Per the checks tab, it seems the APK failed to upload:

```
No files were found with the provided path: $GITHUB_WORKSPACE/oppia.apk. No artifacts will be uploaded.
```

I suppose this one reference might need to be absolute instead of relative.

**Comment by BenHenning** [comment]
LGTM from code perspective. Just waiting to see the upload get fixed

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -130,5 +130,42 @@ jobs:
...
+          unzip bazel-tools.zip
+          tar -xf $HOME/oppia-bazel/android_tools.tar.gz -C $GITHUB_WORKSPACE/tmp/android_tools
+      - name: Unzip Bazel Binary and Build Oppia
+        run: |
+          cd $HOME/oppia-bazel
+          unzip bazel-build.zip
+          cd $GITHUB_WORKSPACE
+          chmod a+x $HOME/oppia-bazel/bazel
+          $HOME/oppia-bazel/bazel build //:oppia --android_databinding_use_v3_4_args --experimental_android_databinding_v2 --override_repository=android_tools=$GITHUB_WORKSPACE/tmp/android_tools --java_header_compilation=false --noincremental_dexing --define=android_standalone_dexing_tool=d8_compat_dx
+          cp $GITHUB_WORKSPACE/bazel-bin/oppia.apk $GITHUB_WORKSPACE
```
Comment (BenHenning) [inline .github/workflows/main.yml]: Maybe copy to the absolute path, too, to make sure it’s correct.

Comment (miaboloix) [reply .github/workflows/main.yml]: Trying that now

**Review by BenHenning — APPROVED** [review]
As mentioned, code LGTM so approved. Please only merge once you verify the APK is available in the list of artifacts.

**Pull request merged**